includes/admin/class-sensei-learners-main.php

```diff
@@ -924,8 +924,13 @@ private function filter_activities_by_users( $search ) {
 	private function is_manually_enrolled( $user_id ) {
 		$enrolment_manager         = Sensei_Course_Enrolment_Manager::instance();
 		$manual_enrolment_provider = $enrolment_manager->get_manual_enrolment_provider();
+		$is_enrolled               = $manual_enrolment_provider->is_enrolled( $user_id, $this->course_id );
 
-		return $manual_enrolment_provider->is_enrolled( $user_id, $this->course_id );
+		// Reset the enrolment state store to avoid out of memory issues.
+		// TODO: This can be removed once the state store no longer holds all the data in memory.
```

Comment:
> once the state store no longer holds all the data in memory.

Do we have an issue to implement this already?

Reply:
Thanks for the review, Renatho! I'm not sure. It's possible that the `Sensei_Enrolment_Provider_State_Store` is storing everything in memory to avoid querying the database multiple times per request. I didn't have the time to investigate all the cases.